src/main/java/com/amazonaws/neptune/auth/NeptuneApacheHttpSigV4Signer.java

```diff
@@ -59,16 +60,59 @@ public class NeptuneApacheHttpSigV4Signer extends NeptuneSigV4SignerBase<HttpUri
      * Create a V4 Signer for Apache Commons HTTP requests.
      *
      * @param regionName             name of the region for which the request is signed
-     * @param awsCredentialsProvider the provider offering access to the credentials used for signing the request
+     * @param v1AwsCredentialProvider the provider offering access to the credentials used for signing the request
      * @throws NeptuneSigV4SignerException in case initialization fails
      */
-    public NeptuneApacheHttpSigV4Signer(
-            final String regionName, final AwsCredentialsProvider awsCredentialsProvider)
+    public NeptuneApacheHttpSigV4Signer(final String regionName,
+                                        final AWSCredentialsProvider v1AwsCredentialProvider)
             throws NeptuneSigV4SignerException {
 
+        super(regionName, v1AwsCredentialProvider);
+    }
+
+    /**
+     * Create a V4 Signer for Apache Commons HTTP requests.
+     *
+     * @param regionName             name of the region for which the request is signed
+     * @param v1AwsCredentialProvider the provider offering access to the credentials used for signing the request
+     * @param serviceName            name of the service name used to sign the requests. Defaults to neptune-db
+     * @throws NeptuneSigV4SignerException in case initialization fails
+     */
+    public NeptuneApacheHttpSigV4Signer(final String regionName,
+                                        final AWSCredentialsProvider v1AwsCredentialProvider,
+                                        final String serviceName) throws NeptuneSigV4SignerException {
+
+        super(regionName, v1AwsCredentialProvider, serviceName);
+    }
+
+    /**
+     * Create a V4 Signer for Apache Commons HTTP requests.
+     *
+     * @param regionName             name of the region for which the request is signed
+     * @param awsCredentialsProvider the provider offering access to the credentials used for signing the request
+     * @throws NeptuneSigV4SignerException in case initialization fails
+     */
+    public NeptuneApacheHttpSigV4Signer(final String regionName,
+                                        final AwsCredentialsProvider awsCredentialsProvider) throws NeptuneSigV4SignerException {
```

Comment:
Why is it called `v1AwsCredentialProvider` above?

Reply:
`AwsCredentialProvider` is CredentialProvider from java sdk v2 and `AWSCredentialProvider` is CredentialProvider from java sdk v1(Aws vs AWS). Difference in naming is subtle that its difficult to distinguish easily. Intentionally named cred provider to be `v1AwsCredentialProvider` for v1 credential provider to make the distinction in the constructors

Reply:
Yo! It took me 10 good seconds to see the difference. Alright, make sense!